src/lp.rs

```diff
@@ -0,0 +1,428 @@
+
+        let album = client
+            .album(album_id.clone(), None)
+            .await
+            .context("fetching album")?;
+        let artists = album
+            .artists
+            .iter()
+            .map(|a| a.name.as_ref())
+            .collect::<Vec<_>>()
+            .join(", ");
+        eprintln!("Album pinged: {} - {} ", &artists, &album.name); // Debug
+        let tracks = client
+            .album_track(album_id, None)
+            .enumerate()
+            .map(|(count, mb)| mb.map(|x| (count, x)))
+            .map_ok(|(count, track)| TrackInfo {
+                number: count,
+                name: track.name.to_string(),
+                duration: track.duration.clone(),
+                uri: track.external_urls.get("spotify").map(|s| s.to_owned()),
+            })
+            .try_collect::<Vec<TrackInfo>>()
+            .await?;
+
+        Ok(LPInfo {
+            playlist: PlaylistInfo::AlbumInfo {
+                id: album.id.to_string(),
+                artist: artists.clone(),
+                name: album.name.to_string(),
+                uri: album.external_urls.get("spotify").map(|s| s.to_owned()),
+            },
+            tracks,
+            started: None,
+        })
+    }
+    async fn from_spotify_playlist_id<C: BaseClient>(
+        client: &C,
+        album_id_str: &str,
+    ) -> anyhow::Result<Self> {
+        let playlist_id = rspotify::model::PlaylistId::from_id(album_id_str)
+            .context("trying to parse playlist ID")?;
+
+        let playlist = client
+            .playlist(playlist_id.clone(), None, None)
+            .await
+            .context("fetching playlist")?;
+
+        eprintln!("Playlist pinged: {}", &playlist.name); // Debug
+        let items = client
+            .playlist_items(playlist_id, None, None)
+            .try_collect::<Vec<PlaylistItem>>()
+            .await?;
+        let tracks = items
+            .iter()
+            .enumerate()
+            .filter_map(|(count, item)| {
+                item.track.as_ref().map(|track| match track {
+                    PlayableItem::Track(track) => TrackInfo {
+                        number: count + 1,
+                        name: track.name.to_string(),
+                        duration: track.duration.clone(),
+                        uri: track
+                            .external_urls
+                            .get("spotify")
+                            .map(|s| s.to_owned()),
+                    },
+                    // Why are there episodes in your playlist?
+                    PlayableItem::Episode(episode) => TrackInfo {
+                        number: count + 1,
+                        name: episode.name.to_string(),
+                        duration: episode.duration.clone(),
+                        uri: episode
+                            .external_urls
+                            .get("spotify")
+                            .map(|s| s.to_owned()),
+                    },
```

Comment:
Could be factored a little I think
```suggestion
                item.track.as_ref().map(|track| {
                    let PlayableItem::Track(FullTrack{name, duration, external_urls, ..}) | PlayableItem::Episode(FullEpisode{name, duration, external_urls, ..}) = track;
                    TrackInfo {
                        number: count + 1,
                        name: name.to_string(),
                        duration: duration.clone(),
                        uri: external_urls
                            .get("spotify")
                            .map(|s| s.to_owned()),
                    }
```

Reply:
Ohh, didn't know about OR patterns, cool, I'm all for this!